examples/crm/src/companies/CompanyActivityIterator.tsx

```diff
@@ -0,0 +1,330 @@
+import {
+    Box,
+    Button,
+    List,
+    ListItem,
+    Skeleton,
+    Stack,
+    Typography,
+} from '@mui/material';
+import { useState } from 'react';
+import { Link, useGetOne, useListContext, useRecordContext } from 'react-admin';
+import {
+    COMPANY_CREATED,
+    CONTACT_CREATED,
+    CONTACT_NOTE_CREATED,
+    DEAL_CREATED,
+} from '../consts';
+import { Activity, Company, Contact, ContactNote, Deal, Sale } from '../types';
+
+export function CompanyActivityIterator() {
+    const company = useRecordContext<Company>();
+    const { data: activity, error, isPending } = useListContext<Activity>();
+    if (!company || isPending || error) return null;
+
+    return (
+        <Box>
+            <List
+                sx={{
+                    '& .MuiListItem-root': {
+                        justifyContent: 'space-between',
+                        gap: 2,
+                    },
+                    '& .MuiListItem-root:not(:first-child)': {
+                        borderTop: '1px solid #f0f0f0',
+                    },
+                }}
+            >
+                {activity.map(activity => (
+                    <ActivityItem
+                        key={activity.id}
+                        activity={activity}
+                        company={company}
+                    />
+                ))}
+            </List>
+        </Box>
+    );
+}
+
+function ActivityItem({
+    activity,
+    company,
+}: {
+    activity: Activity;
+    company: Company;
+}) {
+    switch (activity.type) {
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**nitpick (non-blocking):** ‏

I prefer if statement